skrf/vectorFitting.py

```diff
@@ -0,0 +1,765 @@
+
+        :param j:
+            Column index of the response.
+
+        :param freqs:
+            List of frequencies for the response plot. If None, the sample frequencies of the fitted network are used.
+
+        :return:
+            None
+        """
+
+        if freqs is None:
+            freqs = np.linspace(np.amin(self.network.f), np.amax(self.network.f), 1000)
+
+        mplt.figure()
+        #mplt.scatter(self.network.f, 20 * np.log10(np.abs(self.network.s[:, i, j])), color='r', label='Samples')
+        mplt.scatter(self.network.f, np.abs(self.network.s[:, i, j]), color='r', label='Samples')
+        #mplt.plot(freqs, 20 * np.log10(np.abs(self.get_model_response(i, j, freqs))), color='k', label='Fit')
+        mplt.plot(freqs, np.abs(self.get_model_response(i, j, freqs)), color='k', label='Fit')
+        mplt.xlabel('Frequency (Hz)')
+        #mplt.ylabel('Magnitude (dB)')
+        mplt.ylabel('Magnitude')
+        mplt.legend(loc='best')
+        mplt.title('i={}, j={}'.format(i, j))
+        mplt.tight_layout()
+        mplt.show()
+
+    def plot_pz(self, i, j):
+        """
+        Plots a pole-zero diagram of the fit of the response **S_i,j**.
+
+        :param i:
+            Row index of the response.
+
+        :param j:
+            Column index of the response.
+
+        :return:
+            None
+        """
+
+        i_response = i * self.network.nports + j
+        mplt.figure()
+        mplt.scatter((np.real(self.poles), np.real(self.poles)),
+                     (np.imag(self.poles), -1 * np.imag(self.poles)),
+                     marker='x', label='Pole')
+        mplt.scatter((np.real(self.zeros[i_response]), np.real(self.zeros[i_response])),
+                     (np.imag(self.zeros[i_response]), -1 * np.imag(self.zeros[i_response])),
+                     marker='o', label='Zero')
+        mplt.xlabel('Re{s} (rad/s)')
+        mplt.ylabel('Im{s} (rad/s)')
+        mplt.legend(loc='best')
+        mplt.tight_layout()
+        mplt.show()
+
+    def plot_convergence(self):
+        """
+            Plots the history of the model parameter **d_res** during the iterative pole relocation process of the
+            vector fitting, which should eventually converge to a fixed value. Additionally, the relative change of the
+            maximum singular value of the coefficient matrix **A** are plotted, which serve as a convergence indicator.
+
+        :return:
+            None
+        """
+        mplt.figure()
+        mplt.semilogy(np.arange(np.alen(self.delta_max_history)) + 1, self.delta_max_history, color='darkblue')
+        mplt.xlabel('Iteration step')
+        mplt.ylabel('Max. relative change', color='darkblue')
+        ax2 = mplt.twinx()
+        ax2.plot(np.arange(np.alen(self.d_res_history)) + 1, self.d_res_history, color='orangered')
+        ax2.set_ylabel('Residue', color='orangered')
+        mplt.tight_layout()
+        mplt.show()
+
+    def write_spice_subckt_s(self, file):
```

Comment:
Suggestion: `write_spice_subcircuit_s` unless `subckt` is a conventional name in SPICE (?). It only adds few letter and to me it's clearer

Reply:
`subckt` is a SPICE convention, but it does not hurt to make it more readable by naming it write_spice_subcircuit_s. Done in 5981299be06c6f9ac70790e63e4699ba0363ada6